packages/wdio-cli/src/interface.js

```diff
@@ -55,19 +49,58 @@ export default class WDIOCLInterface extends EventEmitter {
              */
             worker: {}
         }
-        this.clearConsole()
+
+        this.onStart()
+    }
+
+    onStart() {
+        this.log(chalk.bold(`\nExecution of ${chalk.blue(this.totalWorkerCnt)} spec files started at`), this.start.toISOString())
+        if (this.inDebugMode) {
+            this.log(chalk.bgYellow.black('DEBUG mode enabled!'))
+        }
+        if (this.isWatchMode) {
+            this.log(chalk.bgYellow.black('WATCH mode enabled!'))
+        }
+        this.log('')
     }
 
-    clearConsole () {
-        this.display = []
+    onSpecRunning(cid) {
+        this.onJobComplete(cid, this.jobs.get(cid), 0, chalk.bold.cyan('RUNNING'))
+    }
+
+    onSpecRetry(cid, job, retries) {
+        this.onJobComplete(cid, job, retries, chalk.bold.yellow('RETRYING'))
+    }
+
+    onSpecPass(cid, job, retries) {
+        this.onJobComplete(cid, job, retries, chalk.bold.green('PASSED'))
+    }
+
+    onSpecFailure(cid, job, retries) {
+        this.onJobComplete(cid, job, retries, chalk.bold.red('FAILED'))
+    }
+
+    onJobComplete(cid, job, retries, message) {
+        const details = []
+        if (job) {
```

Comment:
Do you suggest that `job` could be not exist?

Reply:
Yeah, there is such test on launcher.test.js, not sure how to reproduce this in real life.
In one of the cases job doesn't exist

Reply:
@CrispusDH this test is failing https://github.com/webdriverio/webdriverio/blob/8de7f1a3b12d97282ed4ee2e25e4c3c9a8940ac1/packages/wdio-cli/tests/launcher.test.js#L105

job is undefined in this test both for old and current interface implementations.
It was added by @bennieswart 3 months ago, not sure how to get such situation in real life.